src/server/render.ts

```diff
@@ -586,7 +586,25 @@ options.vnode = (vnode) => {
         );
       };
     }
+  } else if (typeof vnode.type === "string") {
+    // Work around `preact/debug` string event handler error which
+    // errors when an event handler gets a string. This makes sense
+    // on the client where this is a common vector for XSS. On the
+    // server when the string was not created through concatenation
+    // it is fine. Internally, `preact/debug` only checks for the
+    // lowercase variant.
+    if (vnode.props !== null) {
+      const props = vnode.props as Record<string, unknown>;
+      for (const key in props) {
+        const value = props[key];
+        if (key.startsWith("on") && typeof value === "string") {
+          delete props[key];
+          props["ON" + key.slice(2)] = value;
```

Comment:
```suggestion
          props[key.replace("on", "ON")] = value;
```
Nit: tiny bit clearer what's going on

Reply:
-1 I always get confused if `.replace()` replaces only the first occurrence or more.